File: Scorpio.Conversion.Runtime/CSharp/Example/src/TableTestCsv.cs
//本文件为自动生成，请不要手动修改
using System;
using System.IO;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using Scorpio.Conversion.Runtime;

namespace Datas {
    public partial class TableTestCsv : ITable {
        const string FILE_MD5_CODE = "f07d3fff17de6b37025a951b272f6c4c";
        private int m_count = 0;
        private Dictionary<int, DataTestCsv> m_dataArray = new Dictionary<int, DataTestCsv>();
        public TableTestCsv Initialize(string fileName, IReader reader) {
            var row = reader.ReadInt32();
            var layoutMD5 = reader.ReadString();
            if (layoutMD5 != FILE_MD5_CODE) {
                throw new Exception("File schemas do not match [TableTestCsv] : " + fileName);
            }
            ConversionUtil.ReadHead(reader);
            for (var i = 0; i < row; ++i) {
                var pData = new DataTestCsv(fileName, reader);
                if (m_dataArray.TryGetValue(pData.ID, out var value))
                    value.Set(pData);
                else
                    m_dataArray[pData.ID] = pData;
            }
            m_count = m_dataArray.Count;
            return this;
        }
        public DataTestCsv GetValue(int ID) {
            if (m_dataArray.TryGetValue(ID, out var value))
                return value;
            throw new Exception($"TableTestCsv not found data : {ID}");
        }
        public bool Contains(int ID) {
            return m_dataArray.ContainsKey(ID);
        }
        public Dictionary<int, DataTestCsv> Datas() {
            return m_dataArray;
        }
        public IData GetValueObject(object ID) {
            return GetValue((int)ID);
        }
        public bool ContainsObject(object ID) {
            return Contains((int)ID);
        }
        public IDictionary GetDatas() {
            return Datas();
        }
        public int Count() {
            return m_count;
        }
    }
}

File: Scorpio.Conversion.Runtime/Python/Example/src/TableTestCsv.py
#本文件为自动生成，请不要手动修改

from DataTestCsv import *
class TableTestCsv:
    def __init__(this):
        this.m_count = 0
        this.m_dataArray = {}

    def Initialize(this, fileName, reader):
        row = reader.ReadInt32()
        if "f07d3fff17de6b37025a951b272f6c4c" != reader.ReadString():
            raise Exception("File schemas do not match [TableTestCsv] : " + fileName)
        reader.ReadHead()
        for i in range(0, row):
            pData = DataTestCsv(fileName, reader)
            if this.m_dataArray.__contains__(pData.ID):
                this.m_dataArray[pData.ID].Set(pData)
            else:
                this.m_dataArray[pData.ID] = pData
        this.m_count = len(this.m_dataArray)
        return this

    def GetValue(this, ID):
        return this.m_dataArray[ID]

    def Contains(this, ID):
        return this.m_dataArray.__contains__(ID)

    def Datas(this):
        return this.m_dataArray

    def GetValueObject(this, ID):
        return this.GetValue(ID)

    def ContainsObject(this, ID):
        return this.Contains(ID)

    def Count(this):
        return this.m_count
